app/pages/api/analyst/upload.ts

```diff
@@ -0,0 +1,103 @@
+import { NextApiHandler, NextApiRequest, NextApiResponse } from "next";
+// 👉️ import the Google Cloud client library
+import { Storage } from "@google-cloud/storage";
+// 👉️ import formdata parser library
+import multer from "multer";
+import { Readable } from "stream";
+
+// 📌 IMPORTANT: prevent next from trying to parse the form
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+};
+
+// 👇️ persist file to GCS bucket
+async function saveFormData(file) {
+  // 👇️ pass-in the SECRET KEY from our Service Account
+  // the secret-key file has all the necessary data needed to authenticate with the Google Cloud
+  /*📌 IMPORTANT: 
+  The client libraries support Application Default Credentials (ADC)
+  So, no need to explicitly authenticate or manage tokens; these requirements are managed automatically by the authentication libraries.
+  To set up ADC se: : https://cloud.google.com/docs/authentication/provide-credentials-adc
+  1. Create a service account with the roles our application needs, and a key for that service account
+  2. Set the environment variable GOOGLE_APPLICATION_CREDENTIALS to the (external to the project) path of the JSON file that contains your service account key. 
+  Linux or macOS CLI
+  export GOOGLE_APPLICATION_CREDENTIALS="KEY_PATH"
+  Windows
+  $env:GOOGLE_APPLICATION_CREDENTIALS="KEY_PATH"
+  Replace KEY_PATH with the path of the JSON file that contains your service account key.
+Note: this environment variable only applies to your current shell session, so if you open a new session, set the variable again.
+  */
+  const storage = new Storage({
+    projectId: process.env.GOOGLE_PROJECT_NAME,
+  });
+  // 👇️ GCS bucket
+  const bucket = storage.bucket(process.env.GOOGLE_BUCKET_NAME);
+
+  //check connection
+  // storage.getBuckets().then((x) => console.log(x));
+
+  // 👇️ create stream
+  const readStream = Readable.from(file.buffer);
+  await new Promise((res) =>
+    readStream
+      .pipe(
+        bucket.file(file.originalname).createWriteStream({
+          resumable: true,
+          gzip: false,
+        })
+      )
```

Comment:
Until (🤔 or possibly in place of?) we have a db table to handle the information we want with the file, it might be useful to [add metadata to the upload itself](https://cloud.google.com/nodejs/docs/reference/storage/latest/storage/file#uploading-a-file-with-metadata). It looks like we could just add something like:

```
...createWriteStream({
  ...,
  { metadata: {
    contentType : 'some/type'
    metadata: {
      uploaderName: 'username@address.com',
      uploadDate: Date.Now(),
      uploadTrack: 'csv/waste',
    },
  }
```

An alternative would be to use [setMetadata](https://cloud.google.com/nodejs/docs/reference/storage/latest/storage/file#_google_cloud_storage_File_setMetadata_member_1_) in `app/app/components/route/dataset/Add.tsx:52-55`, though that might mean needing to build another route.

Not something for this PR, more just noting it for the future.

Reply:
Agree, fantastic addition. I will add a note to the ticket so we can implement next sprint.